Source/Network/StreamsAPI.swift

```diff
@@ -18,23 +18,55 @@ class StreamsAPI: BaseAPI {
     
     override class func parseAsset(data: Data) throws -> Asset {
         guard let responseDict = try JSONSerialization.jsonObject(with: data) as? [String: Any],
-              let video = responseDict["video"] as? [String: Any],
               let id = responseDict["id"] as? String,
               let title = responseDict["title"] as? String,
-              let playbackURL = video["playback_url"] as? String,
-              let status = video["status"] as? String,
-              let content_protection_type = video["content_protection_type"] as? String else {
-            throw NSError(domain: "InvalidResponseError", code: 0)
+              let contentType = responseDict["type"] as? String else {
+            throw NSError(domain: "InvalidResponseError", code: 0, userInfo: [NSLocalizedDescriptionKey: "Missing required fields in response"])
         }
-        
+
+        let video = parseVideo(from: responseDict["video"] as? [String: Any])
+        let liveStream = parseLiveStream(from: responseDict["live_stream"] as? [String: Any])
+
         return Asset(
             id: id,
             title: title,
-            video: Video(
-                playbackURL: playbackURL,
-                status: status,
-                drmEncrypted: content_protection_type == "drm"
-            )
+            contentType: contentType,
+            video: video,
+            liveStream: liveStream
+        )
+    }
+    
+    class func parseVideo(from dictionary: [String: Any]?) -> Video? {
+        guard let videoDict = dictionary,
+              let playbackURL = videoDict["playback_url"] as? String,
+              let status = videoDict["status"] as? String,
+              let contentProtectionType = videoDict["content_protection_type"] as? String else {
+            return nil
+        }
+        
+        return Video(
+            playbackURL: playbackURL,
+            status: status,
+            drmEncrypted: contentProtectionType == "drm"
+        )
+    }
```

Comment:
Let this be static method in Video class.
It should be used like Video.parse(dict)

Same for live stream